sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/assessment-workflow-enqueuer.scheduler.ts

```diff
@@ -31,20 +37,37 @@ export class AssessmentWorkflowEnqueuerScheduler extends BaseScheduler<void> {
   @Process()
   async enqueueAssessmentOperations(
     job: Job<void>,
-  ): Promise<ProcessSummaryResult[]> {
+  ): Promise<QueueProcessSummaryResult> {
     const summary = new QueueProcessSummary({
```

Comment:
I am starting to think that, the QueueProcessSummary could be transformed as a JobLoggerService which takes processSummary and logs it. Hence remove all other responsibilities it does.

something like this to explain my statement above better

```ts
const jobLoggerService = new JobLoggerService(job);
const processSummary = new ProcessSummary();
processSummary .info("Checking application assessments to be queued for start.");

try {
      await this.workflowEnqueuerService.enqueueStartAssessmentWorkflows(
        processSummary,
      );
      processSummary .info(
        "All application assessments queued with success, check logs for further details.",
      );
    } catch (error: unknown) {
      processSummary .error(
        `Unexpected error while executing the job check logs for further details. ${parseJSONError(
          error,
        )}`,
      );
    } finally {
      this.logger.logProcessSummary(processSummary);
      await jobLoggerService .logProcessSummary(processSummary);
    }
    await this.cleanSchedulerQueueHistory();
    return processSummary .getSummary();

```

Reply:
The `QueueProcessSummary` is losing its importance and we can consider removing it as you said. I will create a ticket to propose a refactor taking into consideration the new `ProcessSummary`. Does it make sense?

Reply:
Created the below ticket to continue the discussion.
https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/gh/bcgov/sims/2312